src/AeroSharp/DataAccess/MapAccess/Parsers/MapParser.cs

```diff
@@ -29,8 +29,7 @@ internal sealed class MapParser : IMapParser
         if (mapEntry.Value is not TValue value)
         {
             throw new UnexpectedDataFormatException(
-                $"Unable to parse map entry from bin \"{bin}\". Map value is not a {typeof(TValue).FullName}."
-            );
+                $"Unable to parse map entry from bin \"{bin}\". Map value is not a {typeof(TValue).FullName}--It is: {mapEntry.Value.GetType()}");
```

Comment:
Trying to give the message just a bit more detail here, happy to change the formatting/wording if needed.